src/graphapi/engine.cpp

```diff
@@ -32,52 +32,111 @@ namespace miopen {
 
 namespace graphapi {
 
-EngineBuilder& EngineBuilder::setOpGraph(const OpGraph* opGraph)
+GraphPatternExecutor::~GraphPatternExecutor() = default;
+
+size_t GraphExecutorFind20::getWorkspaceSize() const
 {
-    mOpGraph = checkPtr(opGraph);
-    return *this;
+    size_t workspace_size = 0;
+    auto s                = miopenGetSolutionWorkspaceSize(mSolution, &workspace_size);
+    MIOPEN_THROW_IF(s != miopenStatusSuccess, "get solution workspace size failed");
+
+    return workspace_size;
```

Comment:
[R] Exceptions are not free from the performance POV even if not thrown.
```suggestion
    return miopen::deref(mSolution).GetWorkspaceSize();
```
This should be enough, I guess.